File: Day_22/Q87.java
import java.util.*;

class Q87{
    static Scanner sc = new Scanner(System.in);

    //Character Frequency
    static void character_freq(String str){
        HashMap<Character,Integer> mp = new HashMap<>();

        for(int i = 0;i < str.length();i++){
            char ch = str.charAt(i);

            if(mp.containsKey(ch)){
                mp.put(ch,mp.get(ch) + 1);
            }
            else{
                mp.put(ch,1);
            }
        }

        for(char key : mp.keySet()){
            System.out.println(key + " -> " + mp.get(key));
        }
    }

    public static void main(String args[]){
        String str = sc.nextLine();

        character_freq(str);

        sc.close();
    }
}

File: Day_22/Q87.cpp
#include <bits/stdc++.h>
using namespace std;

//Character Frequency
void character_freq(string str){
    unordered_map<char,int> mp;

    for(int i = 0;i < str.length();i++){
        mp[str[i]]++;
    }

    for(auto it : mp){
        cout << it.first << " -> " << it.second << endl;
    }
}

int main(){
    string str;
    getline(cin,str);

    character_freq(str);

    return 0;
}
